icu4c/source/test/intltest/numbertest_parse.cpp

```diff
@@ -408,5 +409,83 @@ void NumberParserTest::testInfiniteRecursion() {
     assertEquals("Unlimited recursion, expected double", -5.0, resultObject.getDouble(status));
 }
 
+void NumberParserTest::test13733_StrictAndLenient() {
+    IcuTestErrorCode status(*this, "test13733_StrictAndLenient");
+
+    static const struct TestCase {
+        const char16_t* inputString;
+        const char16_t* patternString;
+        long expectedStrictParse;
+        long expectedLenientParse;
+    } cases[] = { {u"CA$ 12", u"¤ 0", 12, 12},
+                  {u"CA$12", u"¤0", 12, 12},
+                  {u"CAD 12", u"¤¤ 0", 12, 12},
+                  {u"12 CAD", u"0 ¤¤", 12, 12},
+                  {u"12 Canadian dollars", u"0 ¤¤¤", 12, 12},
+                  {u"$12 ", u"¤¤¤¤0", 12, 12},
+                  {u"12$", u"0¤¤¤¤", 12, 12},
+                  {u"CA$ 12", u"¤0", NULL, 12},
+                  {u"CA$ 12", u"0 ¤¤", NULL, 12},
+                  {u"CA$ 12", u"0 ¤¤¤", NULL, 12},
+                  {u"CA$ 12", u"¤¤¤¤0", NULL, 12},
+                  {u"CA$ 12", u"0¤¤¤¤", NULL, 12},
+                  {u"CA$12", u"¤ 0", NULL, 12},
+                  {u"CA$12", u"¤¤ 0", NULL, 12},
+                  {u"CA$12", u"0 ¤¤", NULL, 12},
+                  {u"CA$12", u"0 ¤¤¤", NULL, 12},
+                  {u"CA$12", u"0¤¤¤¤", NULL, 12},
+                  {u"CAD 12", u"¤0", NULL, 12},
+                  {u"CAD 12", u"0 ¤¤", NULL, 12},
+                  {u"CAD 12", u"0 ¤¤¤", NULL, 12},
+                  {u"CAD 12", u"¤¤¤¤0", NULL, 12},
+                  {u"CAD 12", u"0¤¤¤¤", NULL, 12},
+                  {u"12 CAD", u"¤ 0", NULL, 12},
+                  {u"12 CAD", u"¤0", NULL, 12},
+                  {u"12 CAD", u"¤¤ 0", NULL, 12},
+                  {u"12 CAD", u"¤¤¤¤0", NULL, 12},
+                  {u"12 CAD", u"0¤¤¤¤", NULL, 12},
+                  {u"12 Canadian dollars", u"¤ 0", NULL, 12},
+                  {u"12 Canadian dollars", u"¤0", NULL, 12},
+                  {u"12 Canadian dollars", u"¤¤ 0", NULL, 12},
+                  {u"12 Canadian dollars", u"¤¤¤¤0", NULL, 12},
+                  {u"12 Canadian dollars", u"0¤¤¤¤", NULL, 12},
+                  {u"$12 ", u"¤ 0", NULL, 12},
+                  {u"$12 ", u"¤¤ 0", NULL, 12},
+                  {u"$12 ", u"0 ¤¤", NULL, 12},
+                  {u"$12 ", u"0 ¤¤¤", NULL, 12},
+                  {u"$12 ", u"0¤¤¤¤", NULL, 12},
+                  {u"12$", u"¤ 0", NULL, 12},
+                  {u"12$", u"¤0", NULL, 12},
+                  {u"12$", u"¤¤ 0", NULL, 12},
+                  {u"12$", u"0 ¤¤", NULL, 12},
+                  {u"12$", u"0 ¤¤¤", NULL, 12},
+                  {u"12$", u"¤¤¤¤0", NULL, 12} };
+    for (auto& cas : cases) {
+        UnicodeString inputString(cas.inputString);
+        UnicodeString patternString(cas.patternString);
+        long parsedStrictValue = NULL;
+        long parsedLenientValue = NULL;
```

Comment:
It doesn't make sense to assign `NULL` to an integer type.

Reply:
Changed to `int64_t`